### PR TITLE
argocd-vault-plugin 1.1.2 (new formula)

### PR DESCRIPTION
**Pull request by werne2j**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/argocd-vault-plugin.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+
+    datetime = `date -u +'%Y-%m-%dT%H:%M:%SZ'`
+
+    ldflags = %W[
+      -s -w
+      -X github.com/IBM/argocd-vault-plugin/version.Version=#{version}
+      -X github.com/IBM/argocd-vault-plugin/version.BuildDate=#{datetime}
```
Comment (carlocab) [inline Formula/argocd-vault-plugin.rb:19]:
```suggestion
      -X github.com/IBM/argocd-vault-plugin/version.BuildDate=#{Time.now.utc.strftime("%Y-%m-%dT%H:%M:%SZ")}
```
Let's use this so we don't need to shell out to generate the date.

File: Formula/argocd-vault-plugin.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    revision: "b91d267dfcc69e8a0f029ca5afed585456f537a3"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+
+    datetime = `date -u +'%Y-%m-%dT%H:%M:%SZ'`
+
```
Comment (carlocab) [inline Formula/argocd-vault-plugin.rb:15]:
```suggestion
```
We no longer need this.

File: Formula/argocd-vault-plugin.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+class ArgocdVaultPlugin < Formula
+  desc "Argo CD plugin to retrieve secrets from Secret Management tools"
+  homepage "https://github.com/IBM/argocd-vault-plugin"
+  url "https://github.com/IBM/argocd-vault-plugin.git",
+    tag:      "v1.1.2",
+    revision: "b91d267dfcc69e8a0f029ca5afed585456f537a3"
```
Comment (carlocab) [inline Formula/argocd-vault-plugin.rb:6]:
```suggestion
      tag:      "v1.1.2",
      revision: "b91d267dfcc69e8a0f029ca5afed585456f537a3"
```
Let's align the arguments to `url`.

File: Formula/argocd-vault-plugin.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    datetime = `date -u +'%Y-%m-%dT%H:%M:%SZ'`
+
+    ldflags = %W[
+      -s -w
+      -X github.com/IBM/argocd-vault-plugin/version.Version=#{version}
+      -X github.com/IBM/argocd-vault-plugin/version.BuildDate=#{datetime}
+      -X github.com/IBM/argocd-vault-plugin/version.CommitSHA=#{Utils.git_head}
+    ].join(" ")
+
+    system "go", "build", *std_go_args, "-ldflags", ldflags, "."
```
Comment (carlocab) [inline Formula/argocd-vault-plugin.rb:23]:
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags), "."
```
Let's use the newer syntax for this.

File: Formula/argocd-vault-plugin.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  test do
+    assert_match "This is a plugin to replace <wildcards> with Vault secrets",
+      shell_output("#{bin}/argocd-vault-plugin --help")
+
+    touch testpath/"empty.yaml"
+    assert_match "Error: Must provide a supported Vault Type",
+      shell_output("#{bin}/argocd-vault-plugin generate ./empty.yaml 2>&1", 1)
```
Comment (carlocab) [inline Formula/argocd-vault-plugin.rb:32]: Thanks for adding a proper test!

**Review by carlocab — COMMENTED** [review]
Thanks, @werne2j. A few suggestions -- please squash these changes into your existing commit.

File: Formula/argocd-vault-plugin.rb (hunk 4 above)
Comment (carlocab) [reply Formula/argocd-vault-plugin.rb:23]: Actually, is the `"."` needed? Isn't that the default?

Comment (werne2j) [reply Formula/argocd-vault-plugin.rb:23]: Good call, it should not be needed

**Comment by werne2j** [comment]
@carlocab changes pushed!

File: Formula/argocd-vault-plugin.rb (hunk 5 above)
Comment (chenrui333) [reply Formula/argocd-vault-plugin.rb:30]: Second to it!

**Review by chenrui333 — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]
Thanks for your contribution to Homebrew, @werne2j! 🍻

**Pull request closed without merging**